tests/system/test_apport_retrace.py

```diff
@@ -136,17 +137,34 @@ def _assert_is_retraced(report: Report) -> None:
 
 
 def _assert_divide_by_zero_retrace(report: Report) -> None:
-    stack_top = "divide_by_zero () at /usr/src/chaos-marmosets"
-    if report["DistroRelease"] == "Ubuntu 22.04":
-        stack_top = "divide_by_zero () at ./divide-by-zero.c"
+    # The path could be relative to the source code (with or without
+    # a leading "./" or it can be absolute with /usr/src/<deb-source>-<deb-version>
+
+    # noble: at /usr/src/chaos-marmosets-0.2.0-1/divide-by-zero.c
+    # noble armhf: at divide-by-zero.c
+    # jammy: at ./divide-by-zero.c
+
+    # Actual frame info format depends on arch, compiler, opt-level...
+    # If the failing instruction is *not* the first instruction for the
+    # line there's an addition pointer in the line.
+    frame_regex = re.compile(
+        r"^#0  (0x[0-9a-b]+ in )? divide_by_zero \(\)"
```

Comment:
my bad: I broke the hex-value expression.
```suggestion
        r"^#0  (0x[0-9a-f]+ in )? divide_by_zero \(\)"
```

Reply:
Boooo, bad bdrung ;).
I also spotted another small mistake (a non-escaped `.`) a little further down.

Reply:
Also, I know where that's from. That's actually my original mistake that I fixed back when I created the commits, but that was after our pairing session so your local copy never got the fix.

Reply:
Since frame_regex is a longer version of stack_regex, you might want to deduplicate the code.

Reply:
I thought about it already, but the easy path was thwarted by the `^` and then the dedup code was beginning to be a bit too ugly, so I decided to scrap it.